edb/edgeql/compiler/stmt.py

```diff
@@ -133,10 +133,14 @@ def compile_SelectQuery(
             forward_rptr=forward_rptr,
             ctx=sctx)
 
-        stmt.where = clauses.compile_where_clause(expr.where, ctx=sctx)
+        with sctx.new(context.ContextSwitchMode.NEW) as sctx_no_dml:
+            sctx_no_dml.disallow_dml = True
```

Comment:
I'd do this in `clauses` itself, for two reasons:
 * A new context is already getting made there, so it'll be a bit faster
 * But more importantly, we want this to be true for where clauses on updates and deletes also

Reply:
And I think limit/offset should get checked too

Reply:
I think that DMLs in limits and offsets are fine as they only get executed once.